piccolo/table.py

```diff
@@ -573,6 +573,8 @@ def remove(self) -> Delete:
 
         setattr(self, self._meta.primary_key._meta.name, None)
 
+        self._exists_in_db = False
```

Comment:
In an ideal world we would have a callback on the delete query, so `_exists_in_db` gets set to `False` after the query has completed. But we have no callbacks on delete queries at the moment, so I think this is the best we can do.